BDArmory/VesselSpawning/VesselSpawnerBase.cs

```diff
@@ -876,7 +876,17 @@ protected IEnumerator PlaceSpawnedVessels(List<Vessel> vessels)
         {
             loweringVesselsCount = 0; // Reset the counter for good measure.
             foreach (var vessel in vessels)
-                StartCoroutine(PlaceSpawnedVessel(vessel));
+                if (!(BDArmorySettings.RUNWAY_PROJECT_ROUND == 67 && vessel.GetName().Contains(BDArmorySettings.PINATA_NAME)))
+                    StartCoroutine(PlaceSpawnedVessel(vessel));
+                else
```

Comment:
One issue with performing the HP modification in `ApplyRWP` is that for competitions (other than continuous spawn ones) that is only called for vessels that are in `pilots`, which need to have an AI of some sort (presumably the asteroid piñata doesn't due to the other changes in this PR). I think simply calling `ApplyRWP` in the RWP block at the end of `PostSpawnMainSequence` would be fine though.

Ah right, I missed the logic that `spawnAirborne` was common for all the vessels in `PostSpawnMainSequence` for `CircularSpawning` and `CustomSpawnTemplate`. I'd still rather that this wasn't in the `PlaceSpawnedVessels` function as that means the function isn't necessarily doing what it says on the box. Instead, it's better to filter out the piñata from the list of vessels before calling `PlaceSpawnedVessels`.

A diff that does both of these (and tweaks the piñata warning to include the vessel name(s)) is:
```diff
diff --git a/BDArmory/VesselSpawning/VesselSpawnerBase.cs b/BDArmory/VesselSpawning/VesselSpawnerBase.cs
index 3d6dc04c5..38a0d3d73 100644
--- a/BDArmory/VesselSpawning/VesselSpawnerBase.cs
+++ b/BDArmory/VesselSpawning/VesselSpawnerBase.cs
@@ -443,3 +443,5 @@ protected IEnumerator PostSpawnMainSequence(SpawnConfig spawnConfig, bool spawnA
                 if (BDArmorySettings.DEBUG_SPAWNING) LogMessage("Lowering vessels", false);
-                yield return PlaceSpawnedVessels(spawnedVessels.Values.ToList());
+                var vesselsToPlace = spawnedVessels.Values.ToList();
+                if (BDArmorySettings.RUNWAY_PROJECT && BDArmorySettings.RUNWAY_PROJECT_ROUND == 67) vesselsToPlace = vesselsToPlace.Where(vessel => !vessel.GetName().Contains(BDArmorySettings.PINATA_NAME)).ToList(); // Exclude specific vessels (e.g., some piñatas).
+                yield return PlaceSpawnedVessels(vesselsToPlace);
                 if (spawnFailureReason != SpawnFailureReason.None) yield break;
@@ -465,4 +467,6 @@ protected IEnumerator PostSpawnMainSequence(SpawnConfig spawnConfig, bool spawnA
             SpawnUtils.CheckForRenamedVessels(spawnedVessels);
-            if(spawnedVessels.Any(kvp => kvp.Value.GetName().Contains(BDArmorySettings.PINATA_NAME))) LogMessage("Spawning Pinata", true);
-            if (BDArmorySettings.RUNWAY_PROJECT && !ignoreValidity)
+            foreach (var vessel in spawnedVessels.Values.Where(v => v.GetName().Contains(BDArmorySettings.PINATA_NAME))) LogMessage($"Spawning Piñata: {vessel.GetName()}"); // Warn about spawning piñatas in case of poorly named craft.
+            if (BDArmorySettings.RUNWAY_PROJECT)
+            {
+                if (!ignoreValidity)
                 {
@@ -475,2 +479,4 @@ protected IEnumerator PostSpawnMainSequence(SpawnConfig spawnConfig, bool spawnA
                 }
+                foreach (var vessel in spawnedVessels.Values) SpawnUtils.ApplyRWP(vessel);
+            }
         }
@@ -695,3 +701,3 @@ protected IEnumerator PostSpawnMainSequence(Vessel vessel, bool spawnAirborne, b
             // Lower vessel to the ground or activate them in the air.
-            if (vessel.radarAltitude >= 0 && !spawnAirborne && !(BDArmorySettings.RUNWAY_PROJECT_ROUND == 67 && vessel.GetName().Contains(BDArmorySettings.PINATA_NAME)))
+            if (vessel.radarAltitude >= 0 && !spawnAirborne) // Note: for cts-spawn, spawnAirborne is always true and single-vessel spawning via Aubranium's API is now defunct, so this branch is never taken.
             {
@@ -720,2 +726,3 @@ protected IEnumerator PostSpawnMainSequence(Vessel vessel, bool spawnAirborne, b
                 SpawnUtils.CheckAIWMPlacement(vessel);
+                SpawnUtils.ApplyRWP(vessel);
             }
@@ -881,3 +888,2 @@ protected IEnumerator PlaceSpawnedVessels(List<Vessel> vessels)
             foreach (var vessel in vessels)
-                if (!(BDArmorySettings.RUNWAY_PROJECT_ROUND == 67 && vessel.GetName().Contains(BDArmorySettings.PINATA_NAME)))
                 StartCoroutine(PlaceSpawnedVessel(vessel));
```

Also, the reason to use `!spawnAirborne` instead of `spawnConfig.altitude <= 10` is to avoid hard-coding that altitude limit in multiple places in case we want to adjust it in the future.

Reply:
Huh, github seems to have duplicated this comment...